go/fixchain/fix_and_log.go

```diff
@@ -0,0 +1,99 @@
+package fixchain
+
+import (
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/google/certificate-transparency/go/x509"
+)
+
+// FixAndLog contains a Fixer and a Logger, for all your fix-then-log-chain needs!
+type FixAndLog struct {
+	fixer  *Fixer
+	chains chan []*x509.Certificate
+	logger *Logger
+	wg     sync.WaitGroup
+
+	done        *lockedMap // Cache of chains that QueueAllCertsInChain() has already been called on.
+	alreadyDone uint
+}
+
+// QueueAllCertsInChain adds every cert in the chain and the chain to the queue
+// to be fixed and logged.  Note: chain is expected to be in the order of
+// cert --> root.
```

Comment:
You are right, for the QueueAllCertsInChain it doesn't matter.  For QueueChain it matters that the cert the chain is for comes first for the check to see if it has already been posted, but here as we are fix-then-logging every cert in the chain it doesn't matter.  Comment removed.